weld/codegen/llvm2/mod.rs

```diff
@@ -0,0 +1,1107 @@
+//! An LLVM backend currently optimized for single-threaded execution.
+//!
+//! The `LlvmGenerator` struct is responsible for converting an SIR program into an LLVM module.
+//! This module is then JIT'd and returned into a runnable executable.
+extern crate fnv;
+extern crate time;
+extern crate libc;
+extern crate llvm_sys;
+
+use std::fmt;
+use std::ffi::{CStr, CString};
+
+use fnv::FnvHashMap;
+use libc::{c_char, c_double, c_ulonglong};
+
+use conf::ParsedConf;
+use error::*;
+use sir::*;
+use util::stats::CompilationStats;
+
+use self::llvm_sys::prelude::*;
+use self::llvm_sys::core::*;
+use self::llvm_sys::LLVMLinkage;
+
+use super::*;
+
+static NULL_NAME:[c_char; 1] = [0];
+
+// TODO This should be based on the type!
+pub const LLVM_VECTOR_WIDTH: u32 = 4;
+
+/// Convert a string literal into a C string.
+macro_rules! c_str {
+    ($s:expr) => (
+        concat!($s, "\0").as_ptr() as *const i8
+    );
+}
+
+mod builder;
+mod dict;
+mod intrinsic;
+mod jit;
+mod numeric;
+mod serde;
+mod vector;
+
+use self::builder::merger;
+use self::builder::appender;
+
+pub fn compile(program: &SirProgram,
+               conf: &ParsedConf,
+               stats: &mut CompilationStats,
+               dump_prefix: &str) -> WeldResult<Box<dyn Runnable>> {
```

Comment:
As a general note, it would be good to have a brief comment on every function and struct. Helps a lot with following the code later!